### DISCUSSION TITLE
On my very first and second tries to search on Github, I was blocked, with a message saying I was "rate limited" . Help?  TIA!

### DISCUSSION THREAD

**Discussion by Terry046** [community/community#143228] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Title says it all, I believe :)

**Comment by THE-DEEPDAS** [comment]
Use a GitHub Personal Access Token

For unauthenticated requests, GitHub allows only a small number of searches per hour from the same IP.

If you create a Personal Access Token (PAT) and use it when searching, it will significantly increase your rate limit. Here’s how:

1. Go to GitHub settings.


2. Generate a new token with the necessary permissions.


3. Use the token in your searches, either in command-line tools, scripts, or applications that allow you to authenticate with a PAT.

**Comment by joegeorge022** [comment]
## GitHub’s rate limiting typically happens when multiple requests are sent in a short period. To fix this, you can:

### 1. **Log into GitHub**: Rate limits are higher for logged-in users.
### 2. **Use a Personal Access Token (PAT)**: You can create a PAT under your GitHub account settings. This allows for higher API usage limits if you’re using GitHub's API through code or tools.
### 3. **Avoid multiple searches too quickly**: Try spacing out search attempts a little to avoid triggering the rate limit.




Here’s how to set up a Personal Access Token (PAT) on GitHub:

### 1. Go to GitHub Settings
   - Log in to GitHub and click on your profile icon in the top right corner.
   - Select **Settings** from the dropdown menu.

### 2. Navigate to Developer Settings
   - Scroll down on the left sidebar and click **Developer settings**.
   - Then select **Personal access tokens** > **Tokens (classic)**.

### 3. Generate a New Token
   - Click on **Generate new token**.
   - Enter a **note** (e.g., "GitHub Search PAT") for future reference.

### 4. Set Token Expiration
   - Choose an expiration date or select "No expiration" if you want the token to last indefinitely. For security, GitHub recommends an expiration date.

### 5. Select Scopes
   - To use the token for searching or simple browsing, check **public_repo** and **read:user** under **Select scopes**. These minimal scopes are sufficient if you aren’t doing repository modifications.
   
### 6. Generate and Save the Token
   - Click **Generate token** at the bottom.
   - **Copy the token** right away and save it securely (e.g., in a password manager) as you won’t be able to see it again.

### 7. Use the Token
   - When you’re prompted for authentication on GitHub, enter your GitHub username and use the PAT as the password.
   - For command-line or API requests, use the token with the header `Authorization: token <your-token>`.

This should boost your rate limits significantly, especially if you’re using GitHub’s API or running frequent searches!

**Comment by thiha-min-thant** [comment]
[Primary rate limit for unauthenticated users](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users)
You can make unauthenticated requests if you are only fetching public data. Unauthenticated requests are associated with the originating IP address, not with the user or application that made the request.

The primary rate limit for unauthenticated requests is 60 requests per hour.

[Primary rate limit for authenticated users](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-authenticated-users)
You can use a personal access token to make API requests. Additionally, you can authorize a GitHub App or OAuth app, which can then make API requests on your behalf.

All of these requests count towards your personal rate limit of 5,000 requests per hour. Requests made on your behalf by a GitHub App that is owned by a GitHub Enterprise Cloud organization have a higher rate limit of 15,000 requests per hour. Similarly, requests made on your behalf by a OAuth app that is owned or approved by a GitHub Enterprise Cloud organization have a higher rate limit of 15,000 requests per hour if you are a member of the GitHub Enterprise Cloud organization.

**Comment by MasterInQuestion** [comment]
Some people suggested to open Support tickets..?
    https://github.com/orgs/community/discussions/141073#discussioncomment-11108572
